### ISSUE TITLE
Redirect using a relative URL

### ISSUE THREAD

**Issue by jzheaux** [spring-projects/spring-security#7273]
Spring Security originally redirected on authentication success with an absolute URL, conforming to https://www.rfc-editor.org/rfc/rfc2616#section-14.30. More broadly, any redirects to within the application should use relative URL

However, this RFC is now obsolete. To that end, the reactive stack redirects using a relative URL. It would be good to bring the servlet code into alignment.

**Comment by davidcostanzo** [comment]
When I posted my earlier comment, I was completely lost.  The proposed fix for this issue put me on a path to understanding the control flow.  I think the absolute URL is created in [LoginUrlAuthenticationEntryPoint. buildRedirectUrlToLoginPage()](https://github.com/spring-projects/spring-security/blob/5.0.x/web/src/main/java/org/springframework/security/web/authentication/LoginUrlAuthenticationEntryPoint.java#L178-L207), which goes out of its way to convert a relative URL to an absolute one, conforming to the obsolete RFC.

I was able to work around this problem in my application by creating a subclass of `LoginUrlAuthenticationEntryPoint` that overrides `buildRedirectUrlToLoginPage` to return the URL that `determineUrlToUseForThisRequest` returns, which is a relative URL in my case.  In my application's `SecurityFilterChain` bean, it additionally configures the `HttpSecurity` with

```
   http.exceptionHandling(exceptionHandling -> {
       exceptionHandling.authenticationEntryPoint(new MyLoginUrlAuthenticationEntryPoint("/login"));
   });
```
This ignores any port mapper configuration, so it won't work for everyone.
